File: read-file.js
const fs = require("fs");

async function main() {
    const buffer = (await fs.promises.readFile("unicode.txt"))
    const chars = buffer.toString();
    console.log("buffer size:", buffer.length);
    console.log(buffer);
    for (let i = 0; i < buffer.length; i++) {
        const code = buffer[i];
        const char = String.fromCharCode(code);
        console.log(char + "\t" + code + "\t" + buffer[i].toString(2).padStart(8, "0"));
    }

    console.log("======================================");
    console.log("number of characters:", chars.length);
    console.log(chars);
    for (let i = 0; i < chars.length; i++) {
        const code = chars.charCodeAt(i);
        console.log(chars.charAt(i) + "\t" + code + "\t" + code.toString(2).padStart(16, "0"));
    }

    const realChars = Array.from(chars);
    console.log("======================================");
    console.log(JSON.stringify(realChars));
    console.log("number of real chars:", realChars.length);
    for (let char of realChars) {
        const codePoint = char.codePointAt(0);
        console.log(char + "\t" + codePoint + "\t" + codePoint.toString(2).padStart(32, "0"));
    }

}

main().catch(e => console.log(e.stack));

File: read-file.cc
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

int main() {
    string line;
    ifstream myfile ("unicode.txt");
    getline (myfile,line);
    cout << "number of characters: " << line.size() << endl;
    for (int i = 0; i < line.size(); i++) {
        char code = line[i];
        unsigned char ucode = line[i];
        cout << line[i] << "\t" << to_string(ucode) << "\t" << std::bitset<8>(ucode) << endl;
    }
    cout << line << '\n';
    myfile.close();
}